src/snapshot/src/lib.rs

```diff
@@ -340,6 +340,130 @@ mod tests {
         }
     }
 
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn test_get_format_version_x86() {
+        // Check if `get_format_version()` returns indeed the format
+        // version (the least significant 2 bytes) if the id is valid
+        // (the other bytes == BASE_MAGIC_ID).
+        let mut magic_id = 0x0710_1984_8664_0001u64;
+        assert_eq!(get_format_version(magic_id).unwrap(), 1u16);
+        magic_id = 0x0710_1984_8664_000Eu64;
+        assert_eq!(get_format_version(magic_id).unwrap(), 14u16);
+
+        // Flip a bit to invalidate the arch id.
+        magic_id = 0x1710_1984_8664_0001u64;
+        assert_eq!(
+            get_format_version(magic_id).unwrap_err(),
+            Error::InvalidMagic(magic_id)
+        );
+    }
+
+    #[cfg(target_arch = "aarch64")]
+    #[test]
+    fn test_get_format_version_aarch64() {
+        let mut magic_id = 0x0710_1984_AAAA_0001u64;
+        assert_eq!(get_format_version(magic_id).unwrap(), 1u16);
+        magic_id = 0x0710_1984_AAAA_000Eu64;
+        assert_eq!(get_format_version(magic_id).unwrap(), 14u16);
+        magic_id = 0x1710_1984_AAAA_0001u64;
+        assert_eq!(
+            get_format_version(magic_id).unwrap_err(),
+            Error::InvalidMagic(magic_id)
+        );
+    }
+
+    #[test]
+    fn test_new_snapshot() {
+        let mut vm = VersionMap::new();
+
+        let mut snapshot = Snapshot::new(vm.clone(), 1);
+        assert_eq!(snapshot.format_version, SNAPSHOT_FORMAT_VERSION);
+        assert_eq!(snapshot.target_version, 1);
+        assert_eq!(format!("{:?}", snapshot),
+                   "Snapshot { hdr: SnapshotHdr { data_version: 0, section_count: 0 }, format_version: 1, \
+                    version_map: VersionMap { versions: [{}] }, sections: {}, target_version: 1 }");
+
+        let t1_id = Test1::type_id();
+        vm.new_version().set_type_version(t1_id, 2);
+        snapshot = Snapshot::new(vm.clone(), 2);
+        assert_eq!(snapshot.format_version, SNAPSHOT_FORMAT_VERSION);
+        assert_eq!(snapshot.target_version, 2);
+        assert_eq!(format!("{:?}", snapshot),
+                   format!("Snapshot {{ hdr: SnapshotHdr {{ data_version: 0, section_count: 0 }}, format_version: 1, \
+                    version_map: VersionMap {{ versions: [{{}}, {{{:?}: 2}}] }}, sections: {{}}, target_version: 2 }}", t1_id));
+    }
+
+    #[test]
+    fn test_section_ops() {
+        let vm = VersionMap::new();
```

Comment:
That's a good point, `vmap` sounds good for example.

Reply:
I don't like `vmap` :(. I was actually thinking of letting it as it is because in the snapshotting crates, from what I've seen, we are not making any reference to a virtual machine, so there should be no confusion. But I do not have any strong preference here, I can use `version_map` if you are ok with it.